src/components/QuranReader/TafsirView/TafsirBody.tsx

```diff
@@ -0,0 +1,211 @@
+        </div>
+        <div className={styles.separatorContainer}>
+          <Separator />
+        </div>
+        <div dangerouslySetInnerHTML={{ __html: htmlText }} />
+      </div>
+    );
+  }, []);
+
+  const selectedTafsirLanguage = getSelectedTafsirLanguage(tafsirListData, selectedTafsirId);
+
+  const tafsirContentQuerykey = makeTafsirContentUrl(selectedTafsirId, selectedVerseKey, {
+    words: true,
+    ...getDefaultWordFields(quranReaderStyles.quranFont),
+  });
+
+  return (
+    <div dir={isRTLLanguage(selectedTafsirLanguage) ? 'rtl' : 'ltr'}>
+      <SurahAndAyahLanguageSelection
+        selectedChapterId={selectedChapterId}
+        selectedVerseNumber={selectedVerseNumber}
+        onChapterIdChange={(newChapterId) => {
+          fakeNavigate(
+            getVerseTafsirNavigationUrl(
+              Number(newChapterId),
+              Number(1),
+              selectedTafsirId.toString(),
+            ),
+          );
+          setSelectedChapterId(newChapterId.toString());
+          setSelectedVerseNumber('1'); // reset verse number to 1 every time chapter changes
+        }}
+        onVerseNumberChange={(newVerseNumber) => {
+          setSelectedVerseNumber(newVerseNumber);
+          fakeNavigate(
+            getVerseTafsirNavigationUrl(
+              Number(selectedChapterId),
+              Number(newVerseNumber),
+              selectedTafsirId.toString(),
+            ),
+          );
+        }}
+      />
+      <TafsirAndLanguageSelection
+        selectedTafsirId={selectedTafsirId}
+        selectedLanguage={selectedLanguage}
+        onTafsirSelected={onTafsirSelected}
+        onSelectLanguage={(newLang) => {
+          setSelectedLanguage(newLang);
+        }}
+        languageOptions={languageOptions}
+      />
+      <div
+        className={classNames(
+          styles.tafsirContainer,
+          styles[`tafsir-font-size-${quranReaderStyles.tafsirFontScale}`],
+        )}
+      >
+        {initialTafsirData &&
+        initialChapterId === selectedChapterId &&
+        initialVerseNumber === selectedVerseNumber &&
+        initialTafsirId === selectedTafsirId ? (
+          renderTafsir(initialTafsirData)
+        ) : (
+          <DataFetcher
+            loading={TafsirSkeleton}
+            queryKey={tafsirContentQuerykey}
+            render={renderTafsir}
+          />
+        )}
+      </div>
+    </div>
+  );
+};
+
+/**
+ * Given list of tafsirs, get all available language options
+ *
+ * @param {Tafsir[]} tafsirs
+ * @returns {string[]} list of available language options
+ */
+const getTafsirsLanguageOptions = (tafsirs): string[] =>
+  uniq(tafsirs.map((tafsir) => tafsir.languageName));
+
+export default TafsirBody;
+
+const getSelectedTafsirLanguage = (tafsirListData, selectedTafsirId) => {
+  const selectedTafsir = tafsirListData?.tafsirs.find((tafsir) => tafsir.id === selectedTafsirId);
+  return selectedTafsir?.languageName;
+};
+
+const rtlLanguage = ['arabic', 'urdu'];
```

Comment:
When have a util for this already, we should use the util

Reply:
I think the exisiting utils is for the locale?